apps/frontend/src/app/5_pages/AavePage/components/BorrowPositionsList/components/RepayForm/RepayWithCollateralForm.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { FC, useMemo, useState } from 'react';
+
+import { t } from 'i18next';
+
+import {
+  Button,
+  ErrorBadge,
+  ErrorLevel,
+  HelperButton,
+  SimpleTable,
+  SimpleTableRow,
+} from '@sovryn/ui';
+import { Decimal } from '@sovryn/utils';
+
+import { BOB_CHAIN_ID } from '../../../../../../../config/chains';
+
+import { AmountRenderer } from '../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { AmountTransition } from '../../../../../../2_molecules/AmountTransition/AmountTransition';
+import { AssetAmountInput } from '../../../../../../2_molecules/AssetAmountInput/AssetAmountInput';
+import { AssetRenderer } from '../../../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { config } from '../../../../../../../constants/aave';
+import { useDecimalAmountInput } from '../../../../../../../hooks/useDecimalAmountInput';
+import { translations } from '../../../../../../../locales/i18n';
+import { CollateralRatioHealthBar } from '../../../CollateralRatioHealthBar/CollateralRatioHealthBar';
+
+const pageTranslations = translations.aavePage;
+
+type RepayWithCollateralFormProps = {
+  onComplete: () => void;
+};
+
+export const RepayWithCollateralForm: FC<RepayWithCollateralFormProps> = () => {
+  const assetPrice = 3258.47; // TODO: this is mocked data. Replace with proper hook
+  const totalBorrowed = Decimal.from(10); // TODO: this is mocked data. Replace with proper hook
+  const collateralToLoanRate = Decimal.from(10); // TODO: this is mocked data. Replace with proper hook
+  const collateralSize = Decimal.from(10); // TODO: this is mockd data. Replace with proper hook
+  const assets = useMemo(() => ['BTC', 'SOV'], []); // TODO: this is mocked data. Replace with proper hook
+  const [maximumRepayAmount] = useState(Decimal.from(10)); // TODO: this is mocked data. Replace with proper hook
+  const [maximumRepayWithAmount] = useState(Decimal.from(10)); // TODO: this is mocked data. Replace with proper hook
```

Comment:
this is old, bring new changes

Reply:
Just a remind to check that we will not forget about removing the mock data

Reply:
Yeah, sorry, in the end was not old. This is the paraswap adapter we're not doing, so everything still mocked here, just not accessible so far